src/pstore/pstore.c

```diff
@@ -171,153 +171,146 @@ static int move_file(PStoreEntry *pe, const char *subdir) {
-                 * if files are part of a related group, meaning the same
-                 * original dmesg.
-                 *
-                 * The two known pstore backends are EFI and ERST. These
-                 * backends store data in the Common Platform Error
-                 * Record, CPER, format. The dmesg- filename contains the
-                 * CPER record id, a 64bit number (in decimal notation).
-                 * In Linux, the record id is encoded with two digits for
-                 * the dmesg part (chunk) number and 3 digits for the
-                 * count number. So allowing an additional digit to
-                 * compensate for advancing time, this code ignores the
-                 * last six digits of the filename in determining the
-                 * record id.
-                 *
-                 * For the EFI backend, the record id encodes an id in the
-                 * upper 32 bits, and a timestamp in the lower 32-bits.
-                 * So ignoring the least significant 6 digits has proven
-                 * to generally identify related dmesg entries.  */
-#define PSTORE_FILENAME_IGNORE 6
-
-                /* determine common portion of record id */
-                ++p; /* move beyond dmesg- */
+                ++p; /* move beyond dmesg-xxx- */
                 plen = strlen(p);
-                if (plen > PSTORE_FILENAME_IGNORE) {
-                        pe_id = memdup_suffix0(p, plen - PSTORE_FILENAME_IGNORE);
-                        if (!pe_id) {
+                if (plen < 6)
+                        continue;
+
+                if (startswith(pe->dirent.d_name, "dmesg-efi-")) {
+                        /* For the EFI backend, the 3 least significant digits
+                         * of record id encodes a "count" number, the next 2
+                         * least significant digits for the dmesg part (chunk)
+                         * number, and the remaining digits as the timestamp.
+                         * See linux/drivers/firmware/efi/efi-pstore.c in
+                         * efi_pstore_write(). */
+
+                        /* Extract base record id */
+                        subdir1 = memdup_suffix0(p, plen-5);
+                        if (!subdir1) {
+                                log_oom();
+                                return;
+                        }
+                        /* Extract "count" field */
+                        subdir2 = memdup_suffix0(&p[plen-3], 3);
+                        if (!subdir2) {
                                 log_oom();
                                 return;
                         }
-                } else
-                        pe_id = mfree(pe_id);
 
-                /* Now move file from pstore to archive storage */
-                move_file(pe, pe_id);
+                        /* Now move file from pstore to archive storage */
+                        move_file(pe, subdir1, subdir2);
```

Comment:
Not your fault, but voidify the return code: that is prepend `(void)`.

Reply:
done